Session1.playground/Contents.swift

```diff
@@ -0,0 +1,40 @@
+class Teacher: HasName {
+    func sayName() -> String{
+        return name
+    }
+    
+    var name: String = ""
+    func givingRating(to student: Student, give rating: Int) {
+        student.rating = rating
+    }
+}
+
+protocol HasName {
+    var name: String { get set }
+    func sayName() -> String
+}
+
+class Student: HasName{
+    var name: String = ""
+    var rating = 0
+    
+    func sayName() -> String {
+        return name
+    }
```

Comment:
Можно не писать return
```suggestion
    func sayName() -> String {
       name
    }
```